### ISSUE TITLE
The recognition result is very bad,such as the following iamges

### ISSUE THREAD

**Issue by Jayhello** [bgshih/crnn#37]
![image](https://cloud.githubusercontent.com/assets/16144935/22171252/cef893c2-dfc4-11e6-88f1-7e2dec070eed.png)
result is like that:
Recognized text: vaossyoba (raw: v--a----o-ss-s--y-o-b--a--)

and many simple image ,the result is very bad? anyone can tell me why?

**Comment by bgshih** [comment]
The model is trained on a dataset that is heavily biased towards a-z characters. Training on a balanced dataset would help.

**Comment by rayush7** [comment]
@bgshih If I want the model to recognize words and numbers both then will fine-tuning the model (already trained on VGG Synthetic word dataset) on dataset like Street View House Number (SVHN) gonna help? Or I need to retrain the model from the scratch?

**Comment by bgshih** [comment]
@rayush7 I strongly feel fine-tuning be a better choice than starting from scratch.

**Comment by rayush7** [comment]
@bgshih Thanks. I had another doubt regarding that format of the label of the image. If the image has the word "Apple" in it then its corresponding label would be just a string 'Apple' or a list of string ['A','p','p','l','e'] ? Similarly for images from SVHN datasets. If the image has "350" in it then it's corresponding label would be the string '350' or ['3','5','0'] ? I am really confused because while training I am getting the error in issue#18. (regarding ascii value of the label).

**Comment by rayush7** [comment]
Thanks @bgshih

**Comment by rayush7** [comment]
@bgshih Could you please explain how should the numerical labels be given to the net. 
Example if there is a word "Apple" in the image then setting its groundtruth label as the string 'Apple' is working for me. I can convert itinto lmdb format and create my train and validation set and train my own model.
But if I have numerical labels for example number 350 in the image then by setting its label to string '350' is giving me error in the str2label function (in utilities.lua). Please explain what should be the groundtruth format of the numerical labels?

**Comment by bgshih** [comment]
@rayush7 This is not an expected behavior. What was the error message and where was it thrown?

**Comment by rayush7** [comment]
Actually I am trying to finetune the crnn_demo_model.t7 on SVHN dataset. I am using this command inside crnn/src: 
th main_train.lua ../model/svhn_finetune/ ../model/svhn_finetune/crnn_demo_model.t7

where svhn_finetune is the directory in which the config.lua for finetuning the crnn_demo_model.t7 is present. I have also put a copy of crnn_demo_model.t7, which is the model to be finetuned, in this directory.

The actual error I am getting is : 
/torch/install/bin/luajit: bad argument #3 to '?' (torch.*Tensor expected, got nil)
stack traceback:
[C]: at 0x7f900fc2f8b0
[C]: in function '__newindex'
./utilities.lua:33: in function 'str2label'
./DatasetLmdb.lua:82: in function 'allImageLabel'
./training.lua:84: in function 'trainModel'
main_train.lua:51: in main chunk
[C]: in function 'dofile'
...ning/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
[C]: at 0x004064d0

I tried to print the value of the ascii variable (input to ascii2label function in utilities.lua) and its value is not satisfied by any of the if condition present in the ascii2label function and that is the returned value label is empty. That is why this error appears.

I also tried to print the value of strs variable in str2label function in utilities.lua and instead of showing the list of actual string labels for every image, a list of unreadable string is printed on the console. This is the real cause of the problem. And this is only happening when I use numerical digit as labels (in a string form) for creating my lmdb dataset. That is why I feel there is some issue with creating lmdb dataset for numerical digit as labels (in string form) and not the code in the repository.

Please suggest how do I go about resolving this issue if my explanations resonates with your observation. Sorry for the long reply.

**Comment by rayush7** [comment]
@bgshih Hey the problem was in my lmdb dataset and not in code. I resolved the problem. Now giving the '350' as a label for a image with number in it works like a charm! Thanks @bgshih

**Comment by rayush7** [comment]
@bgshih Could you please suggest how do I plot the learning statistics like the graph between training_loss/validation_loss v # of epochs/# of iterations?

**Comment by rayush7** [comment]
@bgshih I am trying to finetune the crnn model (pretrained on vgg synthetic dataset) on svhn dataset. The images (32x32) I am using look like these
![my_svhn_sample](https://cloud.githubusercontent.com/assets/15009941/25736036/22c6a5c4-318d-11e7-9945-c395dac4a0ac.jpg) ![val_img_99](https://cloud.githubusercontent.com/assets/15009941/25736035/22c14cbe-318d-11e7-942b-30884f7a1a5d.jpg) ![val_img_1668](https://cloud.githubusercontent.com/assets/15009941/25736040/22ce0b52-318d-11e7-9fea-e86a8c265701.jpg) ![val_img_6654](https://cloud.githubusercontent.com/assets/15009941/25736038/22cdc91c-318d-11e7-894e-4a50e4357179.jpg) ![val_img_11111](https://cloud.githubusercontent.com/assets/15009941/25736037/22cad018-318d-11e7-8468-1244aeb12c4f.jpg) ![val_img_20040](https://cloud.githubusercontent.com/assets/15009941/25736039/22cde082-318d-11e7-9c8e-998d0897e561.jpg) ![val_img_23389](https://cloud.githubusercontent.com/assets/15009941/25736042/23043330-318d-11e7-83d8-2526740eeb19.jpg)

While finetuning, the training loss is showing random patterns and no legitimate inference can be drawn because of that. Also all the validation set images are classified as "1". I am unable to figure out if this is a problem due to wrong choice of optimization parameters or crnn is unfit for training on such images. I am using the default config.lua file with adadelta as the optimizer. Any suggestions on what mistake I am making and how can I rectify that?

**Comment by ahmedmazari-dhatim** [comment]
@bgshih can you tell me the steps you followed to fine tune the pre-trained mode with your own dataset. l'm newbie on fine tuning pre-trained model. l want to try it for the first time

**Comment by ahmedmazari-dhatim** [comment]
@bgshih , @rayush7 can you tell me the steps you followed to fine tune the pre-trained mode with your own dataset. l'm newbie on fine tuning pre-trained model. l want to try it for the first time .

Thank you

**Comment by alexiskattan** [comment]
@rayush7 Did you train the model with digits? If so, could you post it online?
